lib/pbench/cli/server/shell.py

```diff
@@ -97,11 +100,18 @@ def run_gunicorn(server_config: PbenchServerConfig, logger: Logger) -> int:
 
     logger.debug("Waiting for database instance to become available.")
     try:
-        Database.wait_for_database(db_uri, db_wait_timeout)
+        wait_for_uri(db_uri, db_wait_timeout)
     except ConnectionRefusedError:
         logger.error("Database {} not responding", db_uri)
         return 1
 
+    logger.debug("Waiting for the Elasticsearch instance to become available.")
+    try:
+        wait_for_uri(es_uri, es_wait_timeout)
+    except ConnectionRefusedError:
+        logger.error("Elasticsearch {} not responding", es_uri)
+        return 1
```

Comment:
Do we want special handling for `BadConfig` here?  (I.e., `ConnectionRefusedError` will result in a relatively quiet exit with a status of `1`, while `BadConfig` will result in an exit with a traceback (and, apparently, an exit status of `1`).)

If not, do we want to change what we're doing for `ConnectionRefusedError`?

Ditto for the call at line 103.

Reply:
Fixed.